version: "3.8"

services:
  mitm-proxy:
    build:
      context: ..
      dockerfile: docker/Dockerfile
    ports:
      - "8080:8080"
      - "8000:8000"
    container_name: mitm-proxy-python
    restart: on-failure
    depends_on:
      - db
    volumes:
      - ../ca.key:/app/ca.key
      - ../ca.crt:/app/ca.crt
      - ../cert.key:/app/cert.key
      - ../init_db.py:/app/init_db.py

  db:
    image: postgres:14
    container_name: mitm-proxy-db
    restart: always
    environment:
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: 1234
      POSTGRES_DB: mitm_proxy
    ports:
      - "5432:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data

  proxy-wrapper:
    build:
      context: ..
      dockerfile: docker/Dockerfile
    container_name: proxy-wrapper
    ports:
      - "8888:8888"
    command: python api/proxy_wrapper.py
    depends_on:
      - mitm-proxy

volumes:
  postgres_data:
